### PR TITLE
Removed pantry pod and codabled everything

### LINKED ISSUE #182: Remove Pantry pod
Depends on #179.

The project uses Pantry for deserialisations and storage. As noted in Pantry's README, it is no longer going to be supported.

So we should probably refactor to use the Codable protocol available in Swift 4. Not a small job.

### PR DESCRIPTION
**Pull request by nickoneill**
Only thing that needs to stick around is a new wrapper struct that we discard. Fixes #182

### CODE REVIEW

**Comment by nickoneill** [comment]
Not entirely sure why the tests are failing on circle, they seem to build OK locally.

**Comment by nickoneill** [comment]
Testing via fastlane fails too. It appears that `NSSearchPathForDirectoriesInDomains(.applicationSupportDirectory, .userDomainMask, true)` isn't a good choice for headless testing.

Not a huge fan of how I use a different path when detecting tests but it seems to work OK. I tried creating a directory at the application support path if it didn't exist but that also failed.

File: FiveCalls/FiveCalls/ContactLog.swift [hunk 1]
```diff
@@ -93,10 +93,16 @@ struct ContactLogs {
...
         let pantryDirName = "com.thatthinginswift.pantry"
         let appSupportDir = NSSearchPathForDirectoriesInDomains(.applicationSupportDirectory, .userDomainMask, true).first!
-        let targetPath = URL(fileURLWithPath: appSupportDir).appendingPathComponent(pantryDirName).appendingPathComponent(ContactLogs.persistenceKey, isDirectory: false)
+
+        let targetPath: URL
+        if UserDefaults.standard.bool(forKey: "isBeingTested") {
+            targetPath = FileManager.default.temporaryDirectory.appendingPathComponent(ContactLogs.persistenceKey, isDirectory: false)
+        } else {
+            targetPath = URL(fileURLWithPath: appSupportDir).appendingPathComponent(pantryDirName).appendingPathComponent(ContactLogs.persistenceKey, isDirectory: false)
+        }
```
Comment (subdigital) [inline FiveCalls/FiveCalls/ContactLog.swift]: Funny enough I did a screencast on how I tend to do this sort of thing:

image

https://nsscreencast.com/episodes/341-detect-testing-in-app-delegate

Comment (nickoneill) [reply FiveCalls/FiveCalls/ContactLog.swift]: Much better - less configuration is better.

**Pull request merged**